core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -740,6 +741,8 @@ private[spark] class ExternalSorter[K, V, C](
           in.close()
         }
       }
+      context.taskMetrics.shuffleWriteMetrics.map(
```

Comment:
Should we write this metric if there was an exception thrown above? if so, maybe we should add this in the finally block. If not, then this is ok.